src/repository.ts

```diff
@@ -59,38 +65,48 @@ export class StorageRepository implements Repository {
     this.cache.del(segmentKey);
   }
 
-  getFlag(identifier: string): FeatureConfig {
+  getFlag(identifier: string, cacheable = true): FeatureConfig {
     const flagKey = this.formatFlagKey(identifier);
     let flag = this.cache.get(flagKey) as FeatureConfig;
     if (flag) {
       return flag;
     }
     if (this.store) {
       flag = this.store.get(flagKey) as FeatureConfig;
-      if (flag) {
+      if (flag && cacheable) {
         this.cache.set(flagKey, flag);
       }
       return flag;
     }
     return undefined;
   }
 
-  getSegment(identifier: string): Segment {
+  getSegment(identifier: string, cacheable = true): Segment {
     const segmentKey = this.formatSegmentKey(identifier);
     let segment = this.cache.get(segmentKey) as Segment;
     if (segment) {
       return segment;
     }
     if (this.store) {
       segment = this.store.get(segmentKey) as Segment;
-      if (segment) {
+      if (segment && cacheable) {
         this.cache.set(segmentKey, segment);
       }
       return segment;
     }
     return undefined;
   }
 
+  private checkFlagVersion(key: string, flag: FeatureConfig): boolean {
```

Comment:
Just personal preference, but I prefer a the function name to describe what it'll give me back. In the case of a boolean like this, something like `isFlagOutdated`